Sources/Confidence/DebugLogger.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+import OSLog
+
+internal protocol DebugLogger {
+    func logEvent(event: ConfidenceEvent, details: String)
+    func logMessage(message: String, isWarning: Bool)
+    func logFlags(flag: String)
+    func logContext(context: ConfidenceStruct)
+}
+
+private extension Logger {
+    private static var subsystem = Bundle.main.bundleIdentifier
+
+    static let confidence = Logger(subsystem: subsystem ?? "", category: "confidence")
+}
+
+internal class DebugLoggerImpl: DebugLogger {
+    func logMessage(message: String, isWarning: Bool) {
+        if !isWarning {
+            Logger.confidence.debug("\(message)")
+        } else {
+            Logger.confidence.warning("\(message)")
+        }
+    }
+
+    func logEvent(event: ConfidenceEvent, details: String) {
```

Comment:
The way `details` is used when calling this API I consider that to be more of an `action`, and I believe it should be just one word, for example `flush`.